File: gui/fgfxFwt/js/FwtToolkitEnvPeer.js
//
// Copyright (c) 2011, chunquedong
// Licensed under the Academic Free License version 3.0
//
// History:
//   2011-7-4  Jed Young  Creation
//

fan.fgfxFwt.FwtToolkitEnvPeer = fan.sys.Obj.$extend(fan.sys.Obj);
fan.fgfxFwt.FwtToolkitEnvPeer.prototype.$ctor = function(self) {}

fan.fgfxFwt.FwtToolkitEnvPeer.initGfxEnv = function() {
   fan.concurrent.Actor.locals().set("fgfx2d.env", new fan.fgfxWtk.GfxEnv());
}

fan.fgfxFwt.FwtToolkitEnvPeer.toGraphics(fan.gfx.Graphics fg) = function() {
  var g = new fan.fgfxWtk.Graphics();
  //g.widget = this;
  g.size = fg.size;
  g.cx = fg.cx;
  g.m_clip = fg.m_clip;

  fg.widget.invalid = function()
  {
    fan.fwt.FwtEnvPeer.$win = this;
    fan.fwt.FwtEnvPeer.$needRelayout = true;
  }
  return g;
}

File: fwtPort/java/fan/fgfxFwt/FwtToolkitEnvPeer.java
//
// Copyright (c) 2011, chunquedong
// Licensed under the Academic Free License version 3.0
//
// History:
//   2011-7-4  Jed Young  Creation
//

package fan.fgfxFwt;

import fan.concurrent.Actor;
import fan.fgfxGraphics.Graphics;
import fan.sys.*;

public class FwtToolkitEnvPeer
{
  public static FwtToolkitEnvPeer make(FwtToolkitEnv self)
  {
    FwtToolkitEnvPeer peer = new FwtToolkitEnvPeer();
    return peer;
  }

  public static void initGfxEnv()
  {
    Actor.locals().set("fgfxGraphics.env", SwtGfxEnv.instance);
  }

  public static Graphics toGraphics(fan.gfx.Graphics fg)
  {
    fan.fwt.FwtGraphics fwt = (fan.fwt.FwtGraphics)fg;
    return new SwtGraphics(fwt.gc());
  }
}